go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -117,6 +118,7 @@ func (vh *vtgateHandler) ComQuery(c *mysql.Conn, query []byte, callback func(*sq
 }
 
 var mysqlListener *mysql.Listener
+var mysqlUnixListener *mysql.Listener
 
 // initiMySQLProtocol starts the mysql protocol.
 // It should be called only once in a process.
```

Comment:
Under there, the next if should probably be:
if *mysqlServerPort < 0  && *mysqlServerSocketPath != "" {
return
}

That way it's possible to have only the socket file, and no port.

Then, line 142 when it says:
// Create a listener
that entire block should be with an:
if *mysqlServerPort >= 0 {
up to the end of the TCP socket stuff.